components/waitwhile/app/waitwhile.app.ts

```diff
@@ -1,13 +1,197 @@
 import { defineApp } from "@pipedream/types";
+import api from "api";
 
 export default defineApp({
   type: "app",
   app: "waitwhile",
-  propDefinitions: {},
+  propDefinitions: {
+    addTag: {
+      label: "Add Tag",
+      type: "string",
+      optional: true,
+      description: "Adds specified tag to existing tags",
+    },
+    customerId: {
+      label: "Customer ID",
+      type: "string",
+      description: "Identifier of customer, automatically derived from visitor contact information if not provided.",
+    },
+    customerNoteId: {
+      label: "Customer Note ID",
+      type: "string",
+      description: "Identifier of customer-note",
+    },
+    desc: {
+      label: "Desc",
+      type: "boolean",
+      optional: true,
+      description: "Show result in descending order",
+    },
+    email: {
+      label: "Email",
+      type: "string",
+      description: "Email address",
+    },
+    externalId: {
+      label: "External Id",
+      type: "string",
+      optional: true,
+      description: "External identifier, optionally provided by client",
+    },
+    firstName: {
+      label: "First Name",
+      type: "string",
+      description: "First name of customer",
+    },
+    fromDate: {
+      label: "From Date",
+      type: "string",
+      optional: true,
+      description: "Date in ISO-8601 format",
+    },
+    fromTime: {
+      label: "From Time",
+      type: "string",
+      optional: true,
+      description: "Date in ISO-8601 format",
+    },
+    inviteId: {
+      label: "Invite ID",
+      type: "string",
+      description: "Unique identifier",
+    },
+    lastName: {
+      label: "Last Name",
+      type: "string",
+      description: "Last name of customer",
+    },
+    limit: {
+      label: "Limit",
+      type: "string",
+      description: "Max number of results",
+    },
+    locationId: {
```

Comment:
Would be nice to add `async options()` for this prop

Reply:
Can a property have more than one async options?  Theses props are passed to several different functions so I would need an async option for each function.